openmdao/core/tests/test_problem.py

```diff
@@ -1128,12 +1128,12 @@ def test_feature_get_set_with_src_indices_diff(self):
         prob.setup()
 
         # set G1.x to 2.0 m, based on the units we gave in the set_input_defaults call
-        prob['G1.x'] = 2.0
+        prob['G1.x'] = np.ones(3) * 2.0
 
         prob.run_model()
 
         # we gave 'G1.x' units of 'm' in the set_input_defaults call
-        assert_near_equal(prob['G1.x'], 2.0, 1e-6)
+        assert_near_equal(prob['G1.x'], np.ones(3) * 2.0, 1e-6)
```

Comment:
Was this a bug before? (i.e., was the target pulling values beyond the bounds of the array?

Reply:
It was a bug, but it wasn't pulling values from beyond the array bounds.  The auto_ivc was the correct size before as well, but the promoted input 'G1.x' was getting mapped to use the src_indices for G1.C1.x so a get_value call was just returning part of the source value.